### ISSUE TITLE
cannot update from 5.1.0 to 5.2.1

### ISSUE THREAD

**Issue by abendt** [kotest/kotest#2873]
**Which version of Kotest are you using**

i am currently using v5.1.0

When trying to update to the latest version 5.2.1 i get some errors when running the tests:

```
        Caused by:
        java.lang.NoClassDefFoundError: kotlinx/coroutines/CopyableThreadContextElement
            at io.kotest.assertions.jvmerrorcollector.getErrorCollectorContextElement(ErrorCollector.kt:20)
            at io.kotest.engine.test.interceptors.CoroutineErrorCollectorInterceptor.intercept(CoroutineErrorCollectorInterceptor.kt:27)
            at io.kotest.engine.test.TestCaseExecutor$execute$2$1.invokeSuspend(TestCaseExecutor.kt:83)
            at io.kotest.engine.test.TestCaseExecutor$execute$2$1.invoke(TestCaseExecutor.kt)
            at io.kotest.engine.test.TestCaseExecutor$execute$2$1.invoke(TestCaseExecutor.kt)
            at io.kotest.engine.test.interceptors.CoroutineDispatcherFactoryInterceptor$intercept$4.invokeSuspend(coroutineDispatcherFactoryInterceptor.kt:57)
            at io.kotest.engine.test.interceptors.CoroutineDispatcherFactoryInterceptor$intercept$4.invoke(coroutineDispatcherFactoryInterceptor.kt)
            at io.kotest.engine.test.interceptors.CoroutineDispatcherFactoryInterceptor$intercept$4.invoke(coroutineDispatcherFactoryInterceptor.kt)
            at io.kotest.engine.concurrency.FixedThreadCoroutineDispatcherFactory$withDispatcher$4.invokeSuspend(FixedThreadCoroutineDispatcherFactory.kt:53)
            at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
            at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
            at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
            at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
            at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by ewoelfel** [comment]
Are you using spring? I had the same error as the wrong version of `io.kotest:kotest-assertions-core-jvm` was pulled (1.5.2).
For me the fix was to add a gradle.properties file containing `kotlin-coroutines.version=1.6.0`

**Comment by abendt** [comment]
thanks @ewoelfel!

yes, i am using spring.
i think you mean `org.jetbrains.kotlinx:kotlinx-coroutines-core-jvm` here?

i have added an explicit dependency to that artifact: `implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core-jvm:1.6.0")`. That helped resolve the issue. Don't understand though why the wrong version was selected in the first place 
🤷🏽‍♂️
thanks again for the hint!

**Comment by jtonic** [comment]
Hi,

I have the same issue with any kotest version greater than `5.1.0` (even the most recent `5.3.0`).
For me the suggestion to use the `implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core-jvm:1.6.0"` didn't work.
The project is indeed a spring boot one... but I do not understand why this happened.
I've tried with kotlinx-coroutines-core-jvm both version 1.6.0 and 1.6.1, without any luck.

The strange thing is that the missing class `kotlinx.coroutines.CopyableThreadContextElement` exists in both coroutines lib versions (1.6.0 and 1.6.1).
Moreover, the only test that exists in the Gradle module is no coroutines specific... so I suppose the root cause is in the kotest itself, or, most probably, a bad combination of some dependencies.

The strange thing is, indeed, that I have no issues with other kotest-based tests in other modules that are no-spring-based, and in which I have a suite of tests with kotest, coroutines, arrow... etc. It is indeed very strange...

I downgrade to version `5.1.0` because of this issue, but I would like to be able to keep with the latest versions (especially because I don't want to have trouble with the IDEA kotest plugin... which I found VERY useful)

Any idea how to address this issue?

Thank you.
